jinja2/environment.py

```diff
@@ -335,6 +336,11 @@ def __init__(self,
         self.enable_async = enable_async
         self.is_async = self.enable_async and have_async_gen
 
+        self.native = native
+        if self.native:
+            self.code_generator_class = NativeCodeGenerator
```

Comment:
Or just make this a completely separate `NativeEnvironment`, rather than an option on the base environment.